service/kernel/data_management/finiteness_checker.cpp

```diff
@@ -0,0 +1,373 @@
+/** file finiteness_checker.cpp */
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "data_management/data/internal/finiteness_checker.h"
+#include "data_management/data/numeric_table.h"
+#include "services/env_detect.h"
+#include "service_dispatch.h"
+#include "threading.h"
+#include "service_numeric_table.h"
+
+namespace daal
+{
+namespace data_management
+{
+namespace internal
+{
+typedef daal::data_management::NumericTable::StorageLayout NTLayout;
+
+void valuesAreFinite(const float * dataPtr, bool * finitenessPrt, size_t n, bool allowNaN)
+{
+    uint32_t * uint32Ptr = (uint32_t *)dataPtr;
+    for (size_t i = 0; i < n; ++i)
+    {
+        // first check: all exponent bits are 1 (inf or nan)
+        if (0x7f800000u == (uint32Ptr[i] & 0x7f800000u))
```

Comment:
create const variables or defines for numbers like `0x7f800000u`
applicable for below

Reply:
Constants are created.